src/box/iproto.cc

```diff
@@ -2322,7 +2322,8 @@ tx_process_call(struct cmsg *m)
 	out = msg->connection->tx.p_obuf;
 	iproto_prepare_select(out, &svp);
 
-	if (msg->header.type == IPROTO_CALL_16)
+	if (msg->header.type == IPROTO_CALL_16 ||
+	    port_c_unwrap_multireturn(&port))
```

Comment:
It seems, we can get rid of iproto.cc changes by adjusting `port_sql_dump_msgpack` function. It is a bit messy, but this mess will be at the lower level.

I see no test fails regarding box.session.push/box_session_push if I change port_c_dump_msgpack directly. Please, re-check this point.

Reply:
Yes. Updated the branch. Also, I renamed functions, as we discussed offline, and added a note about the amount of `box_return_mp()` in the doc request.

`box.session.push()` uses the Lua port, so these changes are unrelated.